グループ機能に関する実装

kotasugimoto opened the pull request:
# what
 - 親子関係の構築
 - groupのedit機能の実装
 - 作成時、編集時にメンバー追加機能の実装
# why
 **アプリケーションに必須の機能のため**

Commit:
other

Commit:
親子関係の構築

Commit:
構造の変更と残り機能の実装

まとまっていてすみません

seoky0 commented:
コミットメッセージは何をしたか明確になるようにおねがします :bow:
otherだと何をしたのか全くわかりませんよね？？
今回はレイアウトの修正・削除なのでそのことがわかるような文言を書きましょう！
コミットから何をしたかわかるように書いていただけるとレビューがとてもしやすくなります！

seoky0 commented on app/controllers/groups_controller.rb:
```diff
@@ -1,26 +1,42 @@
   end
   def create
-    @group = Group.create(group_params)
+    @group = Group.create(create_params)
```
以下でsaveすることになるので、ここでcreateしなくて大丈夫です！
newにとどめておきましょう 🙆

seoky0 commented on app/controllers/groups_controller.rb:
```diff
@@ -1,26 +1,42 @@
-      redirect_to messages_path
+      redirect_to group_messages_path(@group), flash: {notice: 'グループが作成されました。'}
     else
       redirect_to action: 'new'
```
redirect_toではなく、renderの方が再描画には適しています。
詳しく調べてみてください :dog:

seoky0 commented on app/controllers/groups_controller.rb:
```diff
@@ -1,26 +1,42 @@
-  def index
-
+    @group = Group.find(params[:id])
+    @members = @group.members
```
view側で@group.memmbersとすれば良いのでこちらは不要かもしれないです！
確認してみてください。

seoky0 commented on app/controllers/groups_controller.rb:
```diff
@@ -1,26 +1,42 @@
+    @members = @group.members
   end
   def update
+    @group = Group.find(params[:id])
```
editと同じ記述になっているので、メソッドとして定義してbefore_actionで呼び出した方が綺麗になりますね :octocat:

seoky0 commented on app/controllers/groups_controller.rb:
```diff
@@ -1,26 +1,42 @@
   def update
+    @group = Group.find(params[:id])
+    @members = @group.members
+    if @group.update(update_params)
```
良いですね！

seoky0 commented on app/controllers/groups_controller.rb:
```diff
@@ -1,26 +1,42 @@
   end
   private
-  def group_params
+  def update_params
```
create_paramsと同じ記述内容なので、一つあれば良さそうです！
両者で使えるようなメソッドの命名をしてあげてください〜

seoky0 commented on app/controllers/messages_controller.rb:
```diff
@@ -1,21 +1,24 @@
 class MessagesController < ApplicationController
   def index
+    @groups = Group.all
```
こちらの記述は必要でしょうか？？

Commit:
指摘箇所の修正

messages_controllerのindexアクションの＠groupsについての指摘は、allではなくwhereなどで条件指定してグループ
を引っ張ってくるという解釈でよろしいでしょうか？

seoky0 commented:
messages_controllerのindexはあるグループの持つメッセージの一覧なので、Group.allのように複数のグループを取得する必要はないのではないか？という意味でした。

seoky0 commented on app/controllers/groups_controller.rb:
```diff
@@ -7,36 +10,33 @@ def new
       redirect_to group_messages_path(@group), flash: {notice: '登録情報が変更されました。'}
     else
-      redirect_to action: 'new'
+      render 'new'
```
newではなく、editではないでしょうか？

seoky0 commented on app/controllers/groups_controller.rb:
```diff
@@ -1,4 +1,7 @@
+
   def index
     @groups = Group.all
     @message = Message.new
```
グループの一覧ではメッセージを作成することはないはずなので不要ですね。

Commit:
指摘箇所修正

naokishizuka commented on app/controllers/groups_controller.rb:
```diff
@@ -1,24 +1,39 @@
+  def show
 
   end
+  def group_id
```
group_idというメソッド名は実装した本人でなければ
返り値がgroup_idのメソッドであるとミスリードしてしまう可能性があります。
このメソッドの内容は@groupに該当するgroupのインスタンスをセットすることですので
set_groupなどのメソッド名をつけてあげるのが良いでしょう。

naokishizuka commented:
お疲れ様です。一点だけコメントいたしましたのでご確認お願いします。

madana-i commented on app/controllers/application_controller.rb:
```diff
@@ -12,6 +12,6 @@ def after_sign_out_path_for(resource)
 
   def after_sign_in_path_for(resource)
-    '/messages'
+    '/groups'
```
ここはprefix使いましょう

madana-i commented on app/views/messages/index.html.haml:
```diff
@@ -2,237 +2,50 @@
-       サークルメンバー
-
+      = link_to group_messages_path(group),class: "chat-group__link" do
+       = group.group_name
```
groupsテーブルのgroup_nameカラムという命名は、情報が重複しているためよろしくないですね。
groupsテーブルのnameカラムとあれば、そこに入る値はグループの名前であることは明らかなので、変なタイミングで申し訳ありませんがここも修正しましょう！

Commit:
メソッド名変更

madana-i commented on app/views/messages/index.html.haml:
```diff
@@ -2,237 +2,50 @@
-
-     %a(class="chat-group__link" href="")
-
+     - @groups.each do |group|
```
基本的にeachはあまり使わないでパーシャルに切り分けるのが望ましいです！
